### ISSUE TITLE
How to successfully run PowerGenome to create input files for GenX pertaining to only a specific chosen region (in the case below, of WEC_SDGE IPM)

### ISSUE THREAD

**Issue by sambuddhac** [gschivley/PowerGenome#59]
I have been going through the latest PowerGenome README and example notebooks, but, it's still not quite clear to me if I want to generate the inputs for SDGE_IPM region for GenX, how should I proceed. Should I start off from the example_system folder and go inside the test_settings.yml file? I can see over there that the CA_S under region_aggregations is a list that does have the WEC_SDGE as one of the elements. How can I activate only that and not others, so that the input for GenX that I generate (which is the output of PowerGenome) has only data corresponding to WEC-SDGE? Or, is what I am thinking not the right way to proceed? Can you please help me out at your earliest convenience? Thanks a lot !!!!

PS: Latest update: I did the above-mentioned changes (kept only WEC_SDGE within CA_S) and also emptied the CA_N list, as well as commented out the CA_N and WECC_AZ (under model_regions), but, I am getting errors mid-way through, while running the PowerGenome script. 

I tried a few different means of eventually getting rid of all the CA_N entries, wherever it appears in the test_settings.yml and keeping only the WEC_SDGE within CA_S and running PowerGenome each time. I get error at this stage:

[    INFO] powergenome.generators:2363 Results technologies are ['Biomass', 'Hydroelectric Pumped Storage', 'Natural Gas Fired Combined Cycle', 'Natural Gas Fired Combustion Turbine', 'Onshore Wind Turbine', 'Small Hydroelectric', 'Solar Photovoltaic', 'Conventional Hydroelectric', 'Conventional Steam Coal', 'Nuclear']

Here's the error that I get:

Traceback (most recent call last):
  File "/home/samie/anaconda3/envs/powergenome/bin/run_powergenome_multiple", line 33, in <module>
    sys.exit(load_entry_point('powergenome', 'console_scripts', 'run_powergenome_multiple')())
  File "/home/samie/code/PGForked/PowerGenome/powergenome/run_powergenome_multiple_outputs_cli.py", line 233, in main
    gen_clusters = add_misc_gen_values(gen_clusters, _settings)
  File "/home/samie/code/PGForked/PowerGenome/powergenome/GenX.py", line 104, in add_misc_gen_values
    ] = value
  File "/home/samie/anaconda3/envs/powergenome/lib/python3.7/site-packages/pandas/core/indexing.py", line 204, in __setitem__
    indexer = self._get_setitem_indexer(key)
  File "/home/samie/anaconda3/envs/powergenome/lib/python3.7/site-packages/pandas/core/indexing.py", line 182, in _get_setitem_indexer
    return self._convert_tuple(key, is_setter=True)
  File "/home/samie/anaconda3/envs/powergenome/lib/python3.7/site-packages/pandas/core/indexing.py", line 262, in _convert_tuple
    idx = self._convert_to_indexer(k, axis=i, is_setter=is_setter)
  File "/home/samie/anaconda3/envs/powergenome/lib/python3.7/site-packages/pandas/core/indexing.py", line 1278, in _convert_to_indexer
    if com.is_bool_indexer(obj):
  File "/home/samie/anaconda3/envs/powergenome/lib/python3.7/site-packages/pandas/core/common.py", line 130, in is_bool_indexer
    raise ValueError(na_msg)
ValueError: cannot index with vector containing NA / NaN values

**Comment by gschivley** [comment]
Need to fix the function [`add_misc_gen_values`](https://github.com/gschivley/PowerGenome/blob/dc4b1c6e904bbd9286f2f8eaf783da842319b318/powergenome/GenX.py#L91-L106) so that it doesn't look for every resource in the `misc_gen_inputs_fn` file. Or that it can pass over if the resource doesn't exist.

**Comment by gschivley** [comment]
Figured out the real issue in #60

**Comment by gschivley** [comment]
There is also a bug with demand response resources for a single region. The code currently uses `.squeeze` to transform a df into a series, but it turns a series with one row into a value. This parameter is used to map maximum capacity of the DR resource, which causes an error when it only finds a value.

Fixed in 5161fc12d794f234982741c0a8acf1719acdedae on the `bug_fix` branch.

**Comment by gschivley** [comment]
@sambuddhac I fixed the transmission issue, so it doesn't break when no transmission lines exist (51657b4dfb4eb7f93011d9b9d7167eb0d0937523).

I also ran the system with `WEC_SDGE` as the only model region and no region aggregations, which caused a bunch of errors. For now, be sure to comment out the settings parameter `region_aggregations` if you are not doing any aggregations or re-naming. The settings key is now accessed (c9c1aed96ad181171a75ade2e3e31c3f717670e4) using

```py
settings.get("region_aggregations", {})
```

rather than

```py
settings["region_aggregations"]
```

so it sees an empty dictionary rather than `NoneType`.
